include/zephyr/bluetooth/buf.h

```diff
@@ -161,6 +161,40 @@ static inline enum bt_buf_type bt_buf_get_type(struct net_buf *buf)
 		->type;
 }
 
+static inline enum bt_buf_type bt_buf_out_type_from_h4_type(uint8_t h4_type)
```

Comment:
```suggestion
static inline enum bt_buf_type bt_buf_type_from_h4_type(uint8_t h4_type)
```

Why the `out`?

Reply:
I guess because of the `BT_BUF_xx_OUT` mapping. `bt_buf_type` has two of each ACL/ISO H4 types depending on direction.

Reply:
jori is correct. It's not a one-to-one mapping without the direction.

Reply:
I think we should likely get rid of the OUT/IN distinction. IIRC its primary use is for the BTSnoop protocol (CONFIG_BT_MONITOR), however it should be possible to infer it there from the context.

Reply:
> I think we should likely get rid of the OUT/IN distinction. IIRC its primary use is for the BTSnoop protocol (CONFIG_BT_MONITOR), however it should be possible to infer it there from the context.

This PR already makes bt monitor do the inference.

We can deprecate buf types altogether once the transition to H4 encapsulation is finished. We can't make changes to the enum now since that would be a breaking change.

Reply:
> This PR already makes bt monitor do the inference.

Right, sorry, I see that now. I was doing the commenting from the GitHub app on my phone, which isn't the best for getting the "big picture" of PRs :)

> We can deprecate buf types altogether once the transition to H4 encapsulation is finished. We can't make changes to the enum now since that would be a breaking change.

Sounds good.